docs/book/features/orphan_volumes.md

```diff
@@ -0,0 +1,51 @@
+## Orphan volumes detection and clean-up
+Orphan volumes are [FCDs](https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-storage/GUID-AD5AE35E-5209-4775-988C-F86D0E4F0C29.html?hWord=N4IghgNiBcIGYEsBOBnALgAgMYTClGAJgigNYgC+QA) that are present in a vSphere datastore but there is no corresponding PersistentVolume in Kubernetes clusters on the vCenter.
+
+Orphan volumes are often created when CNS solution creates more than one FCD for a Persistent Volume in the Kubernetes cluster. This could occur when the vCenter components are slow, storage is slow, vCenter service restarts, connectivity issues between vCenter and ESXi hosts etc. Since these orphan volumes occupy space in the datastore and are not really used in Kubernetes, it's useful to identify and cleanup orphan volumes periodically.
+
+This functionality provides a set of APIs to detect and delete orphan volumes on-demand, and also provide an option to turn on automatic deletion periodically.
+
+### What qualifies as an orphan volume ?
+An FCD can qualify as an orphan volume if it meets below criteria:
+
+1) A PersistentVolume in any of the registered kubernetes clusters is not using the FCD.
+2) FCD was dynamically provisioned using vSphere CSI driver.
+3) FCD exists for more than 60 minutes(or `vso-create-time-limit-mins` value configured in `cnsmanager-config` configmap) at the time of orphan volumes detection.
+
+### Which orphan volumes are skipped from detection/deletion ?
+Some volumes, even if a kubernetes PersistentVolume doesn't map to them, will not be considered during orphan volume detection/deletion. These include:
+* FCDs created out of band (not using vSphere CSI driver).
+* Statically provisioned CNS volume whose name doesn't start with `pvc-`.
+* File volumes.
+* Orphan volumes that contain snapshots will be detected as orphans but they can not be deleted using Orphan volume delete API.
+
+### A word of caution!
+Orphan FCDs once deleted from a datastore cannot be recovered. It is therefore imperative that you [register all the Kubernetes clusters](../../../README.md#register-kubernetes-clusters-before-you-start) in vCenter with CNS Manager, so that an orphan volume is detected correctly and no crucial data loss happens when they're deleted.
```

Comment:
Is there a way the user can know whether all the k8s clusters are registered with CNS Manager? Just wondering, how can the user trust that the orphan volumes detected by CNS Manager are actually correct and not a misconfiguration?

Reply:
There are no vSphere APIs that we can rely on for this. So currently there's no way to determine this.

Reply:
Although, if a PVC/PV is actively backed by an FCD, the FCD wouldn't get qualified as an Orphan since it would have a corresponding entry in CNS.

So, even if all the clusters aren't registered, the OV detection would work as expected.

Reply:
@kolluria I almost forgot about this validation that we added. I believe we can remove this section then.

Reply:
@gohilankit even though it's not absolutely necessary for the customers to register all the clusters, we should still recommend that they do it.
That way, even if the entry is missing from CNS due to some unforeseen reason(which is highly unlikely), we won't have any surprises. And, our detection algorithm would run much faster if we can find a matching pvc/pv for an fcd.

Reply:
We're providing the recommendation to register all clusters in README.md. Do you think we should keep this section here as well then?

Reply:
I think that should be enough.

Reply:
I renamed the section but still kept it in orphan volumes functionality doc as a reminder to register the clusters.